### PR TITLE
Zenith axis for interpolation as cos(Zenith)

### PR DESCRIPTION
**Pull request by sona-patel**

### CODE REVIEW

File: pyV2DL3/eventdisplay/IrfInterpolator.py [hunk 1]
```diff
@@ -66,13 +66,20 @@ def __load_irf(self):
                 irf_axes[i] = np.concatenate(
                     (axis.flatten(), axis.flatten() + 0.01), axis=None
                 )
-        self.irf_data = irf_data
+
+        irf_axes[-2] = np.cos(np.radians(irf_axes[-2]))[::-1]
+        logging.info(irf_axes[-2])
```
Comment (GernotMaier) [inline pyV2DL3/eventdisplay/IrfInterpolator.py:71]: Not sure what this logging statement does? A printout of values without any additional information is possibly hard to understand.

File: pyV2DL3/eventdisplay/IrfInterpolator.py [hunk 2]
```diff
@@ -66,13 +66,20 @@ def __load_irf(self):
                 irf_axes[i] = np.concatenate(
                     (axis.flatten(), axis.flatten() + 0.01), axis=None
                 )
-        self.irf_data = irf_data
+
+        irf_axes[-2] = np.cos(np.radians(irf_axes[-2]))[::-1]
```
Comment (GernotMaier) [inline pyV2DL3/eventdisplay/IrfInterpolator.py:70]: So the second last axes is always and per definition the zenith axes? Never something else?

**Review by GernotMaier — CHANGES_REQUESTED** [review]
Generally ok - I am just a bit worried about the hard-wired index for coordinates and axes. Are we absolutely sure that this is always correct?

File: pyV2DL3/eventdisplay/IrfInterpolator.py (hunk 2 above)
Comment (sona-patel) [reply pyV2DL3/eventdisplay/IrfInterpolator.py:70]: That's right. The second last axis is always Zenith axis. For 1D irf axes are (energy, pedvar, zenith, camera offsets) and for 2D irf axes are (energy, rad, pedvar, zenith, camera offsets).

Comment (GernotMaier) [reply pyV2DL3/eventdisplay/IrfInterpolator.py:70]: There must be another possibility that we actually read out and check that this is the zenith axis. Nobody will remember this implementation when there is a change in the GDAF. Is there no way to find out what the axis type is?

Comment (sona-patel) [reply pyV2DL3/eventdisplay/IrfInterpolator.py:70]: These axes are read from eventdisplay's irf ROOT file, right? I am not sure if these axes names are stored somewhere! If they are stored then of course we can find out the axis types.

Comment (GernotMaier) [reply pyV2DL3/eventdisplay/IrfInterpolator.py:70]: That's true, it is from the root file. But then you can simply check the axis name.

**Comment by GernotMaier** [comment]
Running now the converter produces a lot of unnecessarily output, could we remove that? It is probably some debug output required during the development? Maybe print only when using the --debug option?

```
INFO:v2dl3: ('IRF axes:', {'energies': array([-1.95, -1.85, -1.75, -1.65, -1.55, -1.45, -1.35, -1.25, -1.15,
       -1.05, -0.95, -0.85, -0.75, -0.65, -0.55, -0.45, -0.35, -0.25,
       -0.15, -0.05,  0.05,  0.15,  0.25,  0.35,  0.45,  0.55,  0.65,
        0.75,  0.85,  0.95,  1.05,  1.15,  1.25,  1.35,  1.45,  1.55,
        1.65,  1.75,  1.85,  1.95,  2.05,  2.15,  2.25,  2.35,  2.45,
        2.55,  2.65,  2.75,  2.85,  2.95,  3.05,  3.15,  3.25,  3.35,
        3.45,  3.55,  3.65,  3.75,  3.85,  3.95], dtype=float32), 'pedvars': array([6.36, 7.  , 7.96], dtype=float32), 'zeniths': array([0.9396926, 1.       ], dtype=float32), 'woffs': array([0.  , 0.25, 0.5 , 0.75, 1.  , 1.25, 1.5 , 1.75, 2.  ],
      dtype=float32)})
INFO:v2dl3: Extracting IRFs of type: hEsysMCRelative2D for azimuth 146.2 deg
INFO:v2dl3: zenith axis index: 3
INFO:v2dl3: ('IRF axes:', {'irf_dimension_1': array([-1.9, -1.7, -1.5, -1.3, -1.1, -0.9, -0.7, -0.5, -0.3, -0.1,  0.1,
        0.3,  0.5,  0.7,  0.9,  1.1,  1.3,  1.5,  1.7,  1.9,  2.1,  2.3,
        2.5,  2.7,  2.9,  3.1,  3.3,  3.5,  3.7,  3.9]), 'irf_dimension_2': array([0.02, 0.06, 0.1 , 0.14, 0.18, 0.22, 0.26, 0.3 , 0.34, 0.38, 0.42,
       0.46, 0.5 , 0.54, 0.58, 0.62, 0.66, 0.7 , 0.74, 0.78, 0.82, 0.86,
       0.9 , 0.94, 0.98, 1.02, 1.06, 1.1 , 1.14, 1.18, 1.22, 1.26, 1.3 ,
       1.34, 1.38, 1.42, 1.46, 1.5 , 1.54, 1.58, 1.62, 1.66, 1.7 , 1.74,
       1.78, 1.82, 1.86, 1.9 , 1.94, 1.98, 2.02, 2.06, 2.1 , 2.14, 2.18,
       2.22, 2.26, 2.3 , 2.34, 2.38, 2.42, 2.46, 2.5 , 2.54, 2.58, 2.62,
       2.66, 2.7 , 2.74, 2.78, 2.82, 2.86, 2.9 , 2.94, 2.98]), 'pedvars': array([6.36, 7.  , 7.96], dtype=float32), 'zeniths': array([0.9396926, 1.       ], dtype=float32), 'woffs': array([0.  , 0.25, 0.5 , 0.75, 1.  , 1.25, 1.5 , 1.75, 2.  ],
      dtype=float32)})
```

**Review by GernotMaier — APPROVED** [review]

**Comment by GernotMaier** [comment]
The CI runs are now successful. @sona-patel  - I don't know what the issue was, but I had to convert the files and my own and upload them. Seems to work now.

**Comment by sona-patel** [comment]
> The CI runs are now successful. @sona-patel - I don't know what the issue was, but I had to convert the files and my own and upload them. Seems to work now.

This is a bit strange! and need to find out why the files processed by two systems are different? I am processing on wgs31.

**Comment by GernotMaier** [comment]
Indeed, we need to test it. I have used this branch, using the effective area and 64080.anasum.root file you get with the downloading scripts, and using the commands as in the CI script.

**Review by GernotMaier — APPROVED** [review]

**Comment by GernotMaier** [comment]
@sona-patel @tobiaskleiner - any reasons not to merge this pull request?

**Comment by sona-patel** [comment]
> @sona-patel @tobiaskleiner - any reasons not to merge this pull request?

No!

**Pull request merged**